src/Xamarin.Android.Build.Tasks/Tasks/Aapt2LinkAssetPack.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+using System.Text.RegularExpressions;
+using System.Collections.Generic;
+using Xamarin.Android.Tools;
+using Microsoft.Android.Build.Tasks;
+
+namespace Xamarin.Android.Tasks {
+
+	public class Aapt2LinkAssetPack : Aapt2 {
+		public override string TaskPrefix => "A2LAP";
+
+		[Required]
+		public ITaskItem Manifest { get; set; }
+
+		[Required]
+		public ITaskItem[] AssetDirectories { get; set; }
+
+		[Required]
+		public string PackageName { get; set; }
+
+		[Required]
+		public ITaskItem OutputArchive { get; set; }
+
+		public string OutputFormat { get; set; } = "proto";
+
+		protected override int GetRequiredDaemonInstances ()
+		{
+			return Math.Min (1, DaemonMaxInstanceCount);
+		}
+
+		public async override System.Threading.Tasks.Task RunTaskAsync ()
+		{
+			RunAapt (GenerateCommandLineCommands (Manifest, OutputArchive), OutputArchive.ItemSpec);
+			ProcessOutput ();
+			if (OutputFormat == "proto" && File.Exists (OutputArchive.ItemSpec)) {
```

Comment:
Potentially (?) user-specifiable strings shouldn't use `==`, but should instead use `string.Compare(OutputFormat, "proto", StringComparison.OrdinalIgnoreCase)`.

Given that this property can't be (easily) set from a `.targets` file, maybe this is moot.  It just compounds with my [previous confusion](https://github.com/xamarin/xamarin-android/pull/8631/files#r1523670702).